### PR TITLE
feat: Negative rotation

### PR DESCRIPTION
**Pull request by andrej8888**
Indicator arrow in drop down menus need arrow animation to revert back instead of cycling full 360 to get to .identity

### CODE REVIEW

**Comment by smajdalf** [comment]
@andrej8888 Pushed working demo to this branch using your code to rotate an element. 

What really means rotate `by180negative`? 

In my case I would expect that animation works like this:
Screen Shot 2021-06-15 at 12 28 52 PM


Your code evokes me:
Screen Shot 2021-06-15 at 12 29 54 PM

The arrow shows the direction of rotation.

**Comment by andrej8888** [comment]
@smajdalf the end result is the same but isSelected ? rotate(.by180) : rotate(.by0)]
Screen Shot 2021-06-15 at 15 07 27
Screen Shot 2021-06-15 at 15 08 27

**Comment by andrej8888** [comment]
https://user-images.githubusercontent.com/78155066/122299360-2dc78e80-cefe-11eb-9cc7-65613330e939.MP4
First varianty without negative rotation

https://user-images.githubusercontent.com/78155066/122299385-36b86000-cefe-11eb-96b6-6658a9e5f4f1.MP4
With negative rotation

**Comment by smajdalf** [comment]
The whole solution with rotation is far from being perfect. Even the previous solution we had. By18Negative would evoke me going from 0 to -180 not from 180 to 0. Even -2 * .pi is equal to 360, so basically, the name of the case is wrong. From the documentation, the whole rotation with positive values is going clockwise. In case you want to rotation counterclockwise, you should use negative values. 

On the other side -> your solution is only working in specific scenarios. Try using this:
1. by90 -> by180Negative
2. by90 -> by0
The behaviour is same, and I think it could be confusing. 

The problem is the whole rotate method. 
Every rotation transformation is animating clockwise or counterclockwise based on the distance it needs to do. 
Basically, it means:

1. by90 (90) -> by180Negative(-360, 0, 360): The nearest is animating counterclockwise 
2. 1. by90 (90) -> by0(-360, 0, 360): The nearest is animating counterclockwise

Why your solution is working? Cause you been lucky and put there valid numbers. Because of this, we should make the case `.custom(degree)` instead of `by180negative`, it would not be confusing but still far from being perfect. We should redesign the whole solution around the rotation process, cause we are lucky that is working.

**Comment by andrej8888** [comment]
@smajdalf I agree but, 
Then whole premise of rotating "by" some degrees fails and is confusing in the case of  isSelected ? rotate(.by180) : rotate(.by0)  because if you rotate by 0 you don't rotate at all. It really should be rotate at
and instead of doing             self?.transform = CGAffineTransform(rotationAngle: CGFloat(rotateBy.rotationValue))
we should do             self?.transform = CGAffineTransform.identify.rotate(rotationAngle: CGFloat(rotateBy.rotationValue)) 
above the identify matrix to achieve the desired rotation even after multiple operations a e should somehow state the direction of rotation as well

File: Sources/GoodExtensions/Array.swift [hunk 1]
```diff
@@ -45,5 +45,14 @@ public extension Array where Element: Equatable {
             append(object)
         }
     }
-    
+
 }
+
+public extension Array {
+
+    var hasItems: Bool {
```
Comment (smajdalf) [inline Sources/GoodExtensions/Array.swift:53]: Missing GRActive extension + deprecation alert as we did before (https://github.com/GoodRequest/GoodIOSExtensions/blob/04b7eb3f84bcc748d83623dd434a2be4391e1421/Sources/GoodExtensions/Array.swift)

File: Sources/GoodExtensions/LossyCodableArray.swift [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+//  sfzfans
+//
+//  Created by Andrej Jasso on 16/06/2021.
+//  Copyright © 2021 GoodRequest. All rights reserved.
+//
+
+import Foundation
+
+@propertyWrapper
+struct LossyCodableArray<Element>: Equatable where Element: Equatable {
```
Comment (smajdalf) [inline Sources/GoodExtensions/LossyCodableArray.swift:12]: Still misssing public access control

**Review by smajdalf — CHANGES_REQUESTED** [review]
